src/__tests__/indexing.ts

```diff
@@ -16,8 +18,8 @@ const DATA2 = { data: 444 }
 const DATA3 = { data: 555 }
 
 declare global {
-  const ceramic: CeramicApi
-  const ceramicClient: CeramicApi
+  const ceramic: Ceramic
```

Comment:
This should probably be `CeramicReaderWriter`, not `Ceramic`.  In most of the test configs this test is going to be using the http-client, not the core client.  I think the only test that should use the `Ceramic` type is the state store test b/c it has to restart the node and only runs in the local_node variant with the core client.

Reply:
I did it this way because `StreamReaderWriter` doesn't have a `did` method:
```
    const originalDid = ceramic.did as DID
```
I can change it to get the DID from the client?
```
    const originalDid = ceramicClient.did as DID
```

Reply:
I reverted it to `Ceramic` because I couldn't get a test to pass, and I don't want to spend time debugging 🤷🏼 That should be a separate ticket, IMO, so I [opened](https://linear.app/3boxlabs/issue/CORE-142/use-ceramic-streamreaderwriter-instead-of-ceramic-ceramic-for-indexing) one.

Reply:
All tests are green now.

Reply:
Wait, it failed the test at runtime?  I literally don't see how changing a typescript type, which disappears at runtime, could possibly change runtime behavior?

I'd like to look into this more, how can I run the tests to reproduce this?

Reply:
Okay I'm assuming the runtime failures were caused by messing with the `did` to work around compile failures.  I put together a branch here that avoids compile failures without doing any messing with the `did`.  Can you check if this works and passes tests, and if so merge into this PR?

Reply:
> Wait, it failed the test at runtime? I literally don't see how changing a typescript type, which disappears at runtime, could possibly change runtime behavior? I would expect it possibly to fail to compile, but not to change behavior at runtime...

Yes, exactly. It wasn't the change itself but likely the fact that I had to remove the line that restored the DID in `afterAll` that resulted in the failure.

> I'd like to look into this more, how can I run the tests to reproduce this?

I was running the tests locally and seeing the same errors in CI.

I'll give the PR a shot, thanks.

Reply:
OK, all tests are passing, thanks, @stbrody.